File: src/main/golang/leetcode/editor/cn/[11]Container With Most Water.go
package lc

//leetcode submit region begin(Prohibit modification and deletion)
func maxArea(height []int) (ans int) {
	Max := func(x, y int) int {
		if x > y {
			return x
		}
		return y
	}
	Min := func(x, y int) int {
		if x > y {
			return y
		}
		return x
	}
	l, r := 0, len(height)-1
	for l < r {
		ans = Max(ans, Min(height[l], height[r])*(r-l))
		if height[l] < height[r] {
			l++
		} else {
			r--
		}
	}
	return ans
}

//leetcode submit region end(Prohibit modification and deletion)

File: src/main/golang/leetcode/editor/cn/[11]Container With Most Water.py
# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def maxArea(self, height: List[int]) -> int:
        n = len(height)
        ans = 0
        l, r = 0, n - 1
        while l <= r:
            h = min(height[l], height[r])
            ans = max(ans, h * (r - l))
            if height[l] <= height[r]:
                l += 1
            else:
                r -= 1
        return ans
# leetcode submit region end(Prohibit modification and deletion)
